src/main/src/main/java/org/vfny/geoserver/global/GeoServerFeatureStore.java

```diff
@@ -95,29 +128,71 @@ public Transaction getTransaction() {
 
     @Override
     public void modifyFeatures(String name, Object attributeValue, Filter filter) throws IOException {
+
+        enforceFeatureUpdateValidation(name, attributeValue);
+
         filter = makeDefinitionFilter(filter);
 
         store().modifyFeatures(name, attributeValue, filter);
     }
 
     @Override
     public void modifyFeatures(String[] names, Object[] attributeValues, Filter filter) throws IOException {
+
+        enforceFeatureUpdateValidation(names, attributeValues);
+
         filter = makeDefinitionFilter(filter);
 
         store().modifyFeatures(names, attributeValues, filter);
     }
 
     @Override
     public void modifyFeatures(Name[] attributeNames, Object[] attributeValues, Filter filter) throws IOException {
+
+        String[] names = Stream.of(attributeNames).map(Name::getLocalPart).toArray(String[]::new);
+        enforceFeatureUpdateValidation(names, attributeValues);
+
         filter = makeDefinitionFilter(filter);
 
         store().modifyFeatures(attributeNames, attributeValues, filter);
     }
 
     @Override
     public void modifyFeatures(Name attributeName, Object attributeValue, Filter filter) throws IOException {
+
+        enforceFeatureUpdateValidation(attributeName.getLocalPart(), attributeValue);
         filter = makeDefinitionFilter(filter);
 
         store().modifyFeatures(attributeName, attributeValue, filter);
     }
+
+    private void enforceFeatureUpdateValidation(String name, Object attributeValue) {
+        enforceFeatureUpdateValidation(new String[] {name}, new Object[] {attributeValue});
+    }
+
+    private void enforceFeatureUpdateValidation(String[] attributeNames, Object[] attributeValues) {
+        for (int i = 0; i < attributeNames.length; i++) {
+            String name = attributeNames[i];
+            Object value = attributeValues[i];
+
+            AttributeDescriptor attributeDescriptor = schema.getDescriptor(name);
+
+            Class<?> binding = attributeDescriptor.getType().getBinding();
+            Object convertedValue = Converters.convert(value, binding);
```

Comment:
One bit here. If one tries to update a numeric value to the value "abc", the conversion will fail, return null, which might even be acceptable for the attribute, and if not, it will result in an odd error message, "attribute xyz cannot be null").

Suggestion: if the value is not null, and the convertedValue is, throw a validation error right away?